ui/plugins/eu.esdihumboldt.hale.ui.functions.groovy/src/eu/esdihumboldt/hale/ui/functions/groovy/internal/TypeStructureTray.java

```diff
@@ -109,6 +110,8 @@ public interface TypeProvider {
 
 	}
 
+	static boolean isMerge = false;
```

Comment:
Static is a bad idea. Like this, once you had one case with a GroovyMergePage this would always be true, even when you edit a Groovy Retype etc.

You could instead pass the variable to the `TypeStructureTray` constructor.